src/main/java/com/google/devtools/build/lib/query2/cquery/LabelAndConfigurationOutputFormatterCallback.java

```diff
@@ -18,39 +18,55 @@
 import com.google.devtools.build.lib.analysis.config.BuildConfiguration;
 import com.google.devtools.build.lib.analysis.config.CoreOptions.IncludeConfigFragmentsEnum;
 import com.google.devtools.build.lib.events.ExtendedEventHandler;
+import com.google.devtools.build.lib.packages.Target;
 import com.google.devtools.build.lib.query2.engine.QueryEnvironment.TargetAccessor;
 import com.google.devtools.build.lib.skyframe.SkyframeExecutor;
 import java.io.OutputStream;
 
 /** Default Output callback for cquery. Prints a label and configuration pair per result. */
 public class LabelAndConfigurationOutputFormatterCallback extends CqueryThreadsafeCallback {
-
+  private final boolean showKind;
   public LabelAndConfigurationOutputFormatterCallback(
       ExtendedEventHandler eventHandler,
       CqueryOptions options,
       OutputStream out,
       SkyframeExecutor skyframeExecutor,
-      TargetAccessor<ConfiguredTarget> accessor) {
+      TargetAccessor<ConfiguredTarget> accessor,
+      boolean showKind
+      ) {
     super(eventHandler, options, out, skyframeExecutor, accessor);
+    this.showKind = showKind;
   }
 
+
   @Override
   public String getName() {
-    return "label";
+    return showKind ? "label_kind" : "label";
   }
 
   @Override
   public void processOutput(Iterable<ConfiguredTarget> partialResult) {
     for (ConfiguredTarget configuredTarget : partialResult) {
       BuildConfiguration config =
           skyframeExecutor.getConfiguration(eventHandler, configuredTarget.getConfigurationKey());
-
-      StringBuilder output =
-          new StringBuilder()
-              .append(configuredTarget.getLabel())
-              .append(" (")
-              .append(config != null && config.isHostConfiguration() ? "HOST" : config)
-              .append(")");
+      StringBuilder output = new StringBuilder();
+      if(showKind){
```

Comment:
Since most of the lines here and in the other branch are repeated, what if `if (showKind)` just sets some string `result` (or some similar name)? Then you only have to write:

    ...
    append(result)

and all those other lines once.